tests.c

```diff
@@ -0,0 +1,137 @@
+#include <locale.h>
+#include <glib.h>
+#include <solv/testcase.h>
+
+#define ARCH     "x86_64"
+#define PLATFORM "f29"
+
+#define ADD_TEST(name, dir) \
+  g_test_add(name, TestData, dir, test_setup, test_run, test_teardown)
+
+typedef struct _test_data {
+  GPtrArray *repos;
+  GStrv solvables;
+  GStrv excluded;
+  gchar *expected;
+} TestData;
+
+extern GPtrArray *fus_depsolve(const char *, const char *, const GStrv, const GStrv, const GStrv, GError **);
+
+static void
+test_run (TestData *td, gconstpointer data)
+{
+  g_autoptr(GError) error = NULL;
+  GStrv repos = (char **)td->repos->pdata;
+
+  g_autoptr(GPtrArray) result = NULL;
+  result = fus_depsolve (ARCH, PLATFORM, NULL, repos, td->solvables, &error);
+  g_assert_no_error (error);
+  g_assert (result != NULL);
+
+  g_autofree char *strres = g_strjoinv ("\n", (char **)result->pdata);
+  g_autofree char *diff = testcase_resultdiff (td->expected, strres);
+  g_assert_cmpstr (diff, ==, NULL);
+}
+
+static void
+test_teardown (TestData *tdata, gconstpointer data)
+{
+  g_ptr_array_unref (tdata->repos);
+  if (tdata->solvables)
+    g_strfreev (tdata->solvables);
+  if (tdata->excluded)
+    g_strfreev (tdata->excluded);
+  if (tdata->expected)
+    g_free (tdata->expected);
+}
+
+static GStrv
+strsplit (char *string, char *delimiter)
```

Comment:
isn't there g_strsplit?

Reply:
Yes, and I use it inside `strsplit`. The problem with just using `g_strsplit` is that it returns an empty string causing fus to print
`fus-WARNING **: 09:54:17.296: Nothing matches ''`

So I created the `strsplit` wrapper to remove '' from the array.

Reply:
This will be unnecessary when #26 is merged, since I can pass the input file itself to the solver instead of its contents.